### PR TITLE
MSP-3191: add webhook protect delete secret mariadb

### PR DESCRIPTION
**Pull request by Uburro**

### CODE REVIEW

File: api/v1/slurmcluster_types.go [hunk 1]
```diff
@@ -441,6 +441,13 @@ type MariaDbOperator struct {
 	// +kubebuilder:validation:Optional
 	Enabled bool `json:"enabled"`
 
+	// If enabled, secret cannot be deleted until custom resource slurmcluster is deleted
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="false"
```
Comment (asteny) [inline api/v1/slurmcluster_types.go:447]: May be it should be true?

Comment (Uburro) [reply api/v1/slurmcluster_types.go:447]: If set to true by default, it will break old clusters where the secret was generated by the MariaDB operator.

Comment (asteny) [reply api/v1/slurmcluster_types.go:447]: But without enabling the option, nobody will use it. May be we should manually recreate secrets for our clients? And write this information to release

Comment (Uburro) [reply api/v1/slurmcluster_types.go:447]: >  nobody will use it. May be we should manually recreate secrets for our clients? And write this information to release

why nobody? we will in new clusters

**Pull request closed without merging**